core/src/main/java/com/linecorp/armeria/server/ServiceRequestContextBuilder.java

```diff
@@ -165,6 +166,15 @@ public ServiceRequestContextBuilder serverConfigurator(Consumer<? super ServerBu
         return this;
     }
 
+    /**
+     * Enable CancellationScheduler option.
+     * See: <a href="https://github.com/line/armeria/issues/5683">line/armeria#5683</a>
+     */
+    public ServiceRequestContextBuilder useCancellationScheduler(boolean useCancellationScheduler) {
+        this.useCancellationScheduler = useCancellationScheduler;
```

Comment:
Can we make the following behavioral change so that `timedOut(true)` works as before:

- Auto-enable `useCancellationScheduler` when `timedOut()` is invoked.
- Disallow disabling `useCancellationScheduler` when `timedOut` is enabled by throwing an `IllegalStateException`.